index.js

```diff
@@ -1,69 +1,50 @@
 'use strict';
-var dns = require('dns');
-var arrify = require('arrify');
-var eachAsync = require('each-async');
-var isPortReachable = require('is-port-reachable');
-var onetime = require('onetime');
-var routerIps = require('router-ips');
-var url = require('url');
-var urlParseLax = require('url-parse-lax');
-
-module.exports = function (dests, cb) {
-	cb = onetime(cb);
-
-	eachAsync(arrify(dests), function (dest, i, done) {
-		dest = urlParseLax(dest);
+const dns = require('dns');
+const url = require('url');
+const arrify = require('arrify');
+const isPortReachable = require('is-port-reachable');
+const routerIps = require('router-ips');
+const urlParseLax = require('url-parse-lax');
+const got = require('got');
+const pify = require('pify');
+
+const checkRedirection = (host, port) => {
+	const protocol = port === 80 ? 'http:' : 'https:';
+
+	return got(host, {protocol}).then(res => {
+		const redirectHost = url.parse(res.headers.location || '').host;
+
+		if (routerIps.has(redirectHost)) {
+			return false;
+		}
+
+		return true;
+	});
+};
 
-		var host = dest.hostname;
-		var port = dest.port || 80;
+module.exports = dests => {
+	return Promise.all(arrify(dests).map(x => {
```

Comment:
This is a bit inefficient as it waits for all the dests to finish. We really only care about the fastest succeeding one. You could use [`p-any`](https://github.com/sindresorhus/p-any) for this.

Same with the browser version.